test/box-luatest/gh_7362_net_box_async_ping_test.lua

```diff
@@ -0,0 +1,36 @@
+local net = require('net.box')
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+local g = t.group()
+
+local TIMEOUT = 5
+
+g.before_all(function()
+    g.server = server:new({alias = 'server'})
+    g.server:start()
+    g.conn = net.connect(g.server.net_box_uri)
+end)
+
+g.after_test('test_connected_ping', function(g)
+    g.server:stop()
+end)
+
+g.after_all(function()
+    g.conn:close()
+end)
+
+g.test_connected_ping = function()
+    t.assert(g.conn:ping({timeout = TIMEOUT}))
+    local future = g.conn:ping({is_async = true})
+    future:wait_result(TIMEOUT)
+    t.assert(future:is_ready())
+    t.assert_equals(future:result(), true)
+end
+
+g.test_disconnected_ping = function(g)
+    t.assert_not(g.conn:ping())
```

Comment:
luatest cases not necessarily run in the same order they were defined. You should stop the server before the test case.

Reply:
Done